gateway/gateway_manager_test.go

```diff
@@ -0,0 +1,24 @@
+package gateway
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestConnecting(t *testing.T) {
+	gtw := NewMockGrpcServer(t)
+	go func() {
+		require.NoError(t, gtw.Serve())
+	}()
+	t.Cleanup(gtw.Stop)
```

Comment:
This is racy. `require.NoError` is evaluated after the test ends. Maybe something like this ensures that the assertion is guaranteed to be part of the test?

```suggestion
	var eg errgroup.Group
	eg.Go(func() error {
		assert.NoError(t, gtw.Serve())
		return nil
	})
	t.Cleanup(func() {
		gtw.Stop()
		eg.Wait()
	})
```